### ISSUE TITLE
WISHLIST: Ability to "mesh" from plane but in 3D

### ISSUE THREAD

**Issue by enzyme69** [chsh2/nijiGPen#22]
I see a huge potential with NijiGPen. So far I have only manage to generate mesh or plane with normal from certain camera ortographic angle.

So I wonder if there is a plan to perhaps using multiple plane canvas (as long grease pencil is planar) -- then ability convert it into flat shapes in 3D.

NOTE: I am thinking of app like Feather3D (web app) and that app is missing some features that Grease Pencil can already do. 

Imagine we are drawing a few strokes in in 3D, but each can generate mesh shapes, potentially like Gravity Sketch.

I use SketchNCarve addon in the past, but I think the NijiGPen Stroke and Fill method to build up shapes, can be super handy if only we can "fill the shape" in 3D space.

**Comment by chsh2** [comment]
Yes, there are plans to add more options for 2D planes besides the orthographic ones. I am thinking about two new modes: viewpoint plane and automatic detection (using [PCA](https://en.wikipedia.org/wiki/Principal_component_analysis) algorithm).

I have done some early verification of these new modes, but such a change will affect almost all operators. Therefore, there is still a lot of work to be done before releasing this feature.

Before finishing, a temporary solution is to rotate the GPencil object (without applying the rotation), then the 2D operation will be performed with the rotated angles instead of the orthographic ones.
